build.zig

```diff
@@ -531,6 +505,21 @@ pub fn build(b: *std.Build) !void {
         .max_rss = 5029889638,
     }));
 
+    test_modules_step.dependOn(tests.addModuleTests(b, .{
+        .test_filters = test_filters,
+        .test_target_filters = test_target_filters,
+        .test_slow_targets = test_slow_targets,
+        .root_src = "src/main.zig",
+        .name = "cli-internals",
```

Comment:
That would work too, so long as it's not misleading since it tests package manager internals and other things not strictly related to compiling code. I had trouble coming up with a suitable name in the first place, the main point I wanted to get across was that it tests `src/main.zig`, similar to how `test-std` tests `lib/std/std.zig`.

Reply:
I'm mostly suggesting `test-compiler-internals` because we already have `test-cli`:

https://github.com/ziglang/zig/blob/e5d9d3f8a128ccf38b7e2f82c6abe70fb34e9eec/test/tests.zig#L1031-L1032

This new step runs all the unit tests defined within the compiler source code, right? In that case, I think `test-cli-internals` would be misleading given the existence of `test-cli` which narrowly tests the CLI.

Reply:
I changed it to `test-compiler-internals`.